core/dbt/contracts/graph/nodes.py

```diff
@@ -497,6 +507,39 @@ def depends_on_nodes(self):
     def depends_on_macros(self):
         return self.depends_on.macros
 
+    def build_contract_checksum(self):
+        # We don't need to construct the checksum if the model does not
+        # have contract enabled, because it won't be used.
+        # This needs to be executed after contract config is set
+        if self.contract is True:
+            contract_state = ""
+            # We need to sort the columns so that order doesn't matter
+            # columns is a str: ColumnInfo dictionary
+            sorted_columns = sorted(self.columns.values(), key=lambda col: col.name)
+            for column in sorted_columns:
+                contract_state += f"|{column.name}"
+                contract_state += column.data_type
+                contract_state += str(column.constraints)
+            data = contract_state.encode("utf-8")
+            self.contract_checksum = hashlib.new("sha256", data).hexdigest()
+
+    def same_contract(self, old) -> bool:
+        if old.contract is False and self.contract is False:
+            # Not a change
+            return True
+        if old.contract is False and self.contract is True:
+            # A change, but not a breaking change
+            return False
+        if old.contract is True and self.contract is False:
+            # Breaking change: throw an error
+            raise (ModelContractError(reason="contract has been disabled", node=self))
+        if self.contract_checksum == old.contract_checksum:
+            # Breaking change: throw an error
+            raise (ModelContractError(reason="column definitions have changed", node=self))
```

Comment:
The reasons for a breaking change can occur at the same time in a state modification check. For example, column definitions changing as well as a contract going from enabled to disabled. Given that these checks are typically used in CI - could we batch up the reasons and raise a single `ModelContractError` with an array of `reasons`? That way the user would have the full picture in their CI logs.